posthog/session_recordings/queries/session_recording_list_from_filters.py

```diff
@@ -99,81 +122,17 @@ def run(self) -> SessionRecordingQueryResult:
         )
 
     def get_query(self):
-        return ast.SelectQuery(
-            select=self._select(),
-            select_from=ast.JoinExpr(table=ast.Field(chain=["raw_session_replay_events"]), alias="s"),
-            where=self._where_predicates(),
-            order_by=[self._order_by_clause()],
-            group_by=[ast.Field(chain=["session_id"])],
-            having=self._having_predicates(),
+        return parse_select(
+            self.BASE_QUERY,
+            {
+                "order_by": self._order_by_clause(),
+                "where_predicates": self._where_predicates(),
+                "having_predicates": self._having_predicates(),
+            },
         )
```

Comment:
i just can't wrap my head around parsing queries this way 🙈 

## with parsing

<img width="1307" alt="with-parsing" src="https://github.com/user-attachments/assets/8d2b664d-006d-4a22-863a-cc24535da00c">

## without parsing

<img width="1301" alt="without-parsing" src="https://github.com/user-attachments/assets/da063926-734e-49e1-8a8f-9e2633db4f79">

so my guess is that the parsing is fast enough that legibility is more important  :)